modules/raft-api/src/main/java/org/apache/ignite/internal/raft/service/RaftGroupListener.java

```diff
@@ -81,4 +82,13 @@ default void onConfigurationCommitted(CommittedConfiguration config) {
      * Invoked once after a raft node has been shut down.
      */
     void onShutdown();
+
+    /**
+     * Invoked before submitting a command to a raft group.
```

Comment:
It seems that you reminded what is performed before writing to the raft log or not?